web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -288,11 +288,24 @@ public Promise<JsTable> getKeyTable() {
         return keys.copy();
```

Comment:
As discussed, we want the `getKeyTable` to be a lazy method that only fetches the key table when needed. Something like:
```java
public Promise<jsTable> getKeyTable() {
  if (keyTable == null) {
    keyTable = connection.newState((c, state, metadata) -> {           
            SelectOrUpdateRequest view = new SelectOrUpdateRequest();
            view.setSourceId(baseTable.state().getHandle().makeTableReference());
            view.setResultId(state.getHandle().makeTicket());
            view.setColumnSpecsList(descriptor.getKeyColumnNamesList());
            connection.tableServiceClient().view(view, metadata, (fail, success) -> {
                keyTable.close();
                c.apply(fail, success);
            });
        }, "view only keys columns")
                .refetch(this, connection.metadata())
                .then(state -> Promise.resolve(new JsTable(connection, state))))
  }
  return keyTable.then(t => t.copy());
}
```

Need to close the `keyTable` promise in the `closeSubscriptions()` method as well.

Reply:
Almost done change here, attempting to resolve issue with types.